EIP-TBD.md

```diff
@@ -0,0 +1,190 @@
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+**Every ERC-TBD compliant contract MUST implement the `ERCTBD` and `ERC165` interfaces** (subject to "caveats" below):
+
+```solidity
+pragma solidity ^0.4.24;
+
+
+/// @title ERCTBDInterface - Access Control Interface
+/// @dev basic inteface for access control mechanism
+/// Note: the ERC-165 identifier for this interface is 0x0b74c80f.
+interface ERCTBDInterface {
+
+    event AssignKey(
+        bytes32 indexed _id,
+        address indexed _from,
+        address indexed _to,
+        bool _assignable,
+        uint80 _startTime,
+        uint80 _expiration,
+        uint80 _uses
+    );
+
+    event RevokeKey(bytes32 indexed _id, address indexed _owner);
+
+    /// @dev transfer partial or all capabilities from the sender to an account
+    /// @param _id lock id
+    /// @param _to recipient
+    /// @param _assignable can the recipient further assign capabilities to other accounts?
+    /// @param _startTime the key's start time (block number)
+    /// @param _expiration the key's expiration time (block number)
+    /// @param _uses number of times this key can be used (in `unlock(..)`)
+    function assignKey(bytes32 _id, address _to, bool _assignable, uint80 _startTime, uint80 _expiration, uint80 _uses) external;
+
+    /// @dev transfer all capabilities from the sender to an account
+    /// @param _id lock id
+    /// @param _to recipient
+    function assignFullKey(bytes32 _id, address _to) external;
+
+    /// @dev revoke the sender's key
+    /// @param _id lock id
+    function revokeKey(bytes32 _id) external;
+
+    /// @dev does the owner have a valid key for the lock id
+    /// @param _id lock id
+    /// @param _owner owner address
+    function unlockable(bytes32 _id, address _owner) external view returns (bool);
+
+    /// @dev does the owner have a valid key for the lock id
+    /// @param _id lock id
+    /// @param _owner owner address
+    /// @return the properties of the requested key as a tuple
+    function getKey(bytes32 _id, address _owner) external view returns (bool, bool, uint80, uint80, uint80);
+}
+
+/// @title ERCTBD - Access Control Interface
+/// @dev contract for access control mechanism
+contract ERCTBD is ERC165, ERCTBDInterface {
+    struct Key {
+        bool exists;
+        bool assignable;
+        uint80 startTime;
```

Comment:
read/write uint80 might cost more than uint256 .

Reply:
How?

Reply:
@ben-kaufman Oren uncovered that it can cost more because of the extra ops needed to extract the individual variables from the word. We ran some tests that suggested its a small difference in `read` (500 gas) vs a large (60,000) difference in `write`. Anyway I suggest we include a detailed calculation of gas usages in the EIP so we can justify our choice (though that can happen in another PR).

Reply:
@dev-matan-tsuberi So you say it makes read cost 500 less because of extraction costs but 60,000 cheaper for writing key because it uses more storage slots, is that correct?
I'm not sure which option is better but I think we should discuss it in another PR.
If you all agree, I'll leave it here as is and then we'll decide what to do and make the relevant changes.